### ISSUE TITLE
[Due for payment 2025-02-13] [$125] Android - Track expense - On deleting track expense, user navigated to expense details page briefly

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#54058]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9. 0.75-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team
**Component** Money Requests

## Action Performed:
1. Launch app
2. Open self dm
3. Create a track expense
4. Tap on header - delete expense

## Expected Result:
On deleting track expense, user must be navigated to self dm

## Actual Result:
On deleting track expense, user navigated to expense details page briefly

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/66e76119-ae37-4715-b7c6-a95b3a24e45b

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869209129305046876</li>
        <li>Upwork Job ID: 1869209129305046876</li>
        <li>Last Price Increase: 2025-01-01</li>
<li>Automatic offers: </li>
<ul>
<li>daledah | Contributor | 105602230</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @twisterdotcom</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Android - Track expense - On deleting track expense, user navigated to expense details page briefly

### What is the root cause of that problem?
When deleting from RHP we dismissModal and navigate to the backRoute when navigation ready here
https://github.com/Expensify/App/blob/50c1db81004285968c6388bc941860f554d411e6/src/libs/ReportUtils.ts#L4233-L4234
but it takes some time in some devices for the navigation to be ready after dismissing modal so the track expense detail page will be visible briefly
### What changes do you think we should make in order to solve the problem?

We should either take the dismissModal into the isNavigationReady so that the details page will not be seen 
```
 Navigation.isNavigationReady().then(() => {
        if (isFromRHP) {
            Navigation.dismissModal();
        }
        Navigation.goBack(backRoute);
    });
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by twisterdotcom** [comment]
Yeah I see this: https://drive.google.com/file/d/1LL2iInslyM3BYoObYJ78Z5fhwdDnql8x/view?usp=sharing

**Comment by twisterdotcom** [comment]
125 because track isn't a priority right now.

**Comment by twisterdotcom** [comment]
We're just waiting for @puneetlath right?

**Comment by twisterdotcom** [comment]
@daledah assigned yesterday.

**Comment by puneetlath** [comment]
FYI the PR for this was reverted: https://github.com/Expensify/App/pull/56381

**Comment by twisterdotcom** [comment]
I also can't recreate. I think we should simply close here I'm afraid.

**Comment by twisterdotcom** [comment]
Payment Summary:

- @dukenv0307 paid $125 via NewDot Manual Request
- @daledah paid $125 [here](https://www.upwork.com/nx/wm/offer/105602230) (Contributor)

**Comment by JmillsExpensify** [comment]
$125 approved for @dukenv0307
